### DISCUSSION TITLE
No SLF4J providers were found - how to add a provider?

### DISCUSSION THREAD

**Discussion by Phillipus** [eclipse-jgit/jgit#22] [Q&A] [answered]
Hi,

we have an Eclipse-base RCP application that uses JGit as a plug-in. When JGit is instantiated the following appears in the console:

```
SLF4J(W): No SLF4J providers were found.
SLF4J(W): Defaulting to no-operation (NOP) logger implementation
```

How do I add a SLF4J provider that JGit can use? I tried adding `slfj4.simple` to the classpath and to the dependencies in our app, but that isn't working.

**Comment by Phillipus** [comment, marked as answer]
It seems that extra stuff has to be added to the product.

See https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/1094
See https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/1226
See https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/commit/b9cb4799d5d0e49129e93c2c3523731a1bba1d94

The changes required can be seen [in this commit](https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/commit/b9cb4799d5d0e49129e93c2c3523731a1bba1d94).

Edit: `startLevel="2"` is required. See [here](https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/commit/bb805d3d70f78137cfa89571bd76ff469dd1ce73)

The full file is [here](https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/master/eclipse.platform.releng.tychoeclipsebuilder/eclipse.platform.repository/platform.product).

1. Edit your product file
2. Add two entries to the Configuration Start Levels:
`<plugin id="org.apache.aries.spifly.dynamic.bundle" autoStart="true" startLevel="2" />`
`<plugin id="slf4j.simple" autoStart="true" startLevel="2" />`
3. Optionally add `-Dorg.slf4j.simpleLogger.defaultLogLevel=off` to the Launching VM Arguments

**Comment by iloveeclipse** [comment]
@Phillipus : could you please add a **summary** of what is missing / what need to be done, because it is almost impossible to get a straight answer looking on longer threads in the list above?

**Comment by Phillipus** [comment]
> @Phillipus : could you please add a **summary** of what is missing / what need to be done, because it is almost impossible to get a straight answer looking on longer threads in the list above?

I've edited the original post.

**Reply by iloveeclipse** [reply]
Perfect, thanks!
